group.go

```diff
@@ -53,11 +53,18 @@ type Scalar interface {
 	// Set to a fresh random or pseudo-random scalar
 	Pick(rand cipher.Stream) Scalar
 
-	// SetBytes will take bytes and create a scalar out of it
+	// SetBytes sets the scalar from a big-endian byte-slice,
+	// reducing if necessary to the appropriate modulus.
 	SetBytes([]byte) Scalar
 
-	// Bytes returns the raw internal representation
+	// Bytes returns a big-Endian representation of the scalar
 	Bytes() []byte
+
+	// Allow or disallow use of faster variable-time implementations
+	// of operations on this Point, returning the old flag value.
+	// This flag always defaults to false (constant-time only)
+	// in implementations that can provide constant-time operations.
+	SetVarTime(varTime bool) bool
```

Comment:
I thought of this approach, and I'm open to it but still on the fence about it.  The main downsides I see are:

* It makes constant-time-Scalars and variable-time-Scalars (for the same Suite) different and incompatible by default, in that you can't for example add a variable-time Scalar to a constant-time Scalar and such without doing awkward conversions.  The need for constant-time operations is often typically specific to particular values - i.e., whether that particular value involves secret information or not - and a lot of calculations in practice involve a combination of secret and non-secret values.  Potentially mitigating this downside somewhat, it might be the case that most of the non-secret values we work with tend to be Points and not Scalars - but if so, that would suggest just making Scalar arithmetic constant-time-only and not bothering to provide faster variable-time versions at all.

* The other functions in the same Suite, particular point.Mul, need to know how to detect and deal with two different types of scalar types: i.e., point.Mul will have to do a type-switch test whether the scalar that was passed as the parameter is actually a scalarVarTime or a scalarConstTime.  Not a big deal and perfectly doable, but a slight cost in complexity and perhaps time (type tests tend to be more expensive than just checking a bool flag, though probably not significant compared to the total cost of a scalar multiply).